### ISSUE TITLE
'Cache is corrupted' warning during metadata generation

### ISSUE THREAD

**Issue by dermeister0** [dotnet/docfx#1626]
### Title
'Cache is corrupted' warning during metadata generation

### Functional impact
Cache is not used.

### Minimal repro steps
1. Clone https://github.com/dermeister0/InheritdocTest repository.
2. Go to \src\Domain directory.
3. Execute `docfx metadata Domain.csproj` command 2 times.

```
PS C:\Work\InheritdocTest\src\Domain> docfx metadata Domain.csproj
[17-05-04 04:19:55.941]Warning:No metadata is generated for Domain.
[17-05-04 04:19:55.942]Info:Completed executing in 1588.9052 milliseconds.


Build succeeded with warning.
[17-05-04 04:19:55.948]Warning:No metadata is generated for Domain.
        1 Warning(s)
        0 Error(s)
PS C:\Work\InheritdocTest\src\Domain> docfx metadata Domain.csproj
[17-05-04 04:20:00.751]Warning:Cache for C:/Work/InheritdocTest/src/Domain/Domain.csproj in C:\Work\InheritdocTest\src\Domain\obj\xdoc\cache\final\-1176272033 is corrupted
[17-05-04 04:20:00.908]Info:'C:/Work/InheritdocTest/src/Domain/Domain.csproj' keep up-to-date since '5/4/2017 4:19:55 AM', cached intermediate result 'C:/Work/InheritdocTest/src/Domain/obj/xdoc/cache/obj\mu4h1pni.xrm' is used.
[17-05-04 04:20:01.043]Warning:No metadata is generated for Domain.
[17-05-04 04:20:01.106]Info:Completed executing in 1394.5891 milliseconds.


Build succeeded with warning.
[17-05-04 04:20:01.112]Warning:Cache for C:/Work/InheritdocTest/src/Domain/Domain.csproj in C:\Work\InheritdocTest\src\Domain\obj\xdoc\cache\final\-1176272033 is corrupted
[17-05-04 04:20:01.112]Warning:No metadata is generated for Domain.
        2 Warning(s)
        0 Error(s)
```

### Expected result
Cache is generated during first build and reused on the second attempt.

### Actual result
'Cache is corrupted' warning is displayed.

### Further technical details
I assume it happens because docfx.json config does not exist and some default values are added to BuildInfo.

![2017-05-04_1101](https://cloud.githubusercontent.com/assets/4054844/25690485/1e86ef90-30bc-11e7-8ec0-04824d3e1f80.png)

**Comment by FranklinTimmy** [comment]
Been facing this since I upgraded to Visual Studio 2017 and upgraded my projects. Though I do understand that the new *csproj* file format still requires some re-configuration in `docfx.json` file. I wanted to check if I could successfully generate API docs for the advertised Seed Project on DocFX portal.

Currently using **Version 2.17.4**

Tried to confirm this error by testing API Metadata generation for the seed project and at first compile got the following.

**On First Run**

```
>H:\docfx-test>docfx.exe
[17-05-14 08:47:10.558]Info:Config file docfx.json found, start generating metadata...
[17-05-14 08:47:12.864]Warning:No metadata is generated for CatLibrary,BaseClass,SomeVbClass.
[17-05-14 08:47:12.964]Info:5 plug-in(s) loaded.
[17-05-14 08:47:12.980]Info:Post processor ExtractSearchIndex loaded.
[17-05-14 08:47:12.995]Info:No files are found with glob pattern **/*.yml, excluding <none>, under directory "H:\docfx-test\obj\api"
[17-05-14 08:47:13.033]Info:Markdown engine is dfm
[17-05-14 08:47:13.049]Info:[Build Document]Max parallelism is 4.
               [17-05-14 08:47:13.634]Info:[Build Document.CompilePhaseHandler.ResourceDocumentProcessor]Building 1 file(s) in ResourceDocumentProcessor(ValidateResourceMetadata)...
[17-05-14 08:47:13.634]Info:[Build Document.CompilePhaseHandler.TocDocumentProcessor]Building 3 file(s) in TocDocumentProcessor(BuildTocDocument)...
[17-05-14 08:47:13.634]Info:[Build Document.CompilePhaseHandler.ManagedReferenceDocumentProcessor]Building 2 file(s) in ManagedReferenceDocumentProcessor(BuildManagedReferenceDocument=>ValidateManagedReferenceDocumentMetadata=>ApplyOverwriteDocumentForMref=>FillReferenceInformation)...
[17-05-14 08:47:13.634]Info:[Build Document.CompilePhaseHandler.RestApiDocumentProcessor]Building 4 file(s) in RestApiDocumentProcessor(BuildRestApiDocument=>ValidateRestApiDocumentMetadata=>ApplyOverwriteDocumentForRestApi)...
[17-05-14 08:47:13.634]Info:[Build Document.CompilePhaseHandler.ConceptualDocumentProcessor]Building 5 file(s) in ConceptualDocumentProcessor(BuildConceptualDocument=>CountWord=>ValidateConceptualDocumentMetadata)...
[17-05-14 08:47:13.665]Warning:[Build Document.CompilePhaseHandler.TocDocumentProcessor.Prebuild.BuildTocDocument](toc.yml)Unable to find either toc.yml or toc.md inside obj/api/. Make sure the file is included in config file docfx.json!
[17-05-14 08:47:13.982]Info:[Build Document.CompilePhaseHandler.ManagedReferenceDocumentProcessor.Build.BuildManagedReferenceDocument](specs/petstore-operations.md#L1)Invalid yaml header.
[17-05-14 08:47:14.467]Info:[Build Document.CompilePhaseHandler.RestApiDocumentProcessor.Build.BuildRestApiDocument](specs/petstore-operations.md#L1)Invalid yaml header.
[17-05-14 08:47:14.922]Info:[Build Document.LinkPhaseHandler.Apply Templates]Applying templates to 11 model(s)...
[17-05-14 08:47:15.222]Info:[Build Document]XRef map exported.
[17-05-14 08:47:15.319]Info:[Postprocess.HandlePostProcessors.Processing ExtractSearchIndex]Extracting index data from 7 html files
[17-05-14 08:47:15.354]Info:[Postprocess]Manifest file saved to manifest.json.
[17-05-14 08:47:15.354]Info:Completed building documents in 2378.8095 milliseconds.
[17-05-14 08:47:15.354]Info:[Apply Theme]Theme is applied.
[17-05-14 08:47:15.354]Info:Completed executing in 4620.3636 milliseconds.


Build succeeded with warning.
[17-05-14 08:47:15.369]Warning:No metadata is generated for CatLibrary,BaseClass,SomeVbClass.
[17-05-14 08:47:15.369]Warning:[Build Document.CompilePhaseHandler.TocDocumentProcessor.Prebuild.BuildTocDocument](toc.yml)Unable to find either toc.yml or toc.md inside obj/api/. Make sure the file is included in config file docfx.json!
        2 Warning(s)
        0 Error(s)

H:\docfx-test>pause
Press any key to continue . . .
```

**On Second Run**

```

H:\docfx-test>docfx.exe
[17-05-14 08:57:51.485]Info:Config file docfx.json found, start generating metadata...
                                                                          [17-05-14 08:57:53.003]Warning:Cache for H:/docfx-test/src/SampleClass1/SampleClass1.sln in H:\docfx-test\src\SampleClass1\obj\xdoc\cache\final\1563805426 is corrupted
[17-05-14 08:57:53.558]Info:'H:/docfx-test/src/SampleClass1/CatLibrary/CatLibrary.csproj' keep up-to-date since '14-May-17 8:47:12 PM', cached intermediate result 'H:/docfx-test/src/SampleClass1/CatLibrary/obj/xdoc/cache/obj\y4hlfywj.als' is used.
[17-05-14 08:57:53.672]Info:'H:/docfx-test/src/SampleClass1/BaseClass/BaseClass.csproj' keep up-to-date since '14-May-17 8:47:12 PM', cached intermediate result 'H:/docfx-test/src/SampleClass1/BaseClass/obj/xdoc/cache/obj\545qmsni.eez' is used.
[17-05-14 08:57:53.672]Info:'H:/docfx-test/src/SampleClass1/SomeVbClass/SomeVbClass.vbproj' keep up-to-date since '14-May-17 8:47:12 PM', cached intermediate result 'H:/docfx-test/src/SampleClass1/SomeVbClass/obj/xdoc/cache/obj\vt5lfyhm.ytb' is used.
[17-05-14 08:57:53.687]Warning:No metadata is generated for CatLibrary,BaseClass,SomeVbClass.
[17-05-14 08:57:53.825]Info:5 plug-in(s) loaded.
[17-05-14 08:57:53.841]Info:Post processor ExtractSearchIndex loaded.
[17-05-14 08:57:53.857]Info:No files are found with glob pattern **/*.yml, excluding <none>, under directory "H:\docfx-test\obj\api"
[17-05-14 08:57:53.888]Info:Markdown engine is dfm
[17-05-14 08:57:53.926]Info:[Build Document]Max parallelism is 4.
[17-05-14 08:57:54.599]Info:[Build Document.CompilePhaseHandler.ResourceDocumentProcessor]Building 1 file(s) in ResourceDocumentProcessor(ValidateResourceMetadata)...
[17-05-14 08:57:54.600]Info:[Build Document.CompilePhaseHandler.ConceptualDocumentProcessor]Building 5 file(s) in ConceptualDocumentProcessor(BuildConceptualDocument=>CountWord=>ValidateConceptualDocumentMetadata)...
[17-05-14 08:57:54.600]Info:[Build Document.CompilePhaseHandler.ManagedReferenceDocumentProcessor]Building 2 file(s) in ManagedReferenceDocumentProcessor(BuildManagedReferenceDocument=>ValidateManagedReferenceDocumentMetadata=>ApplyOverwriteDocumentForMref=>FillReferenceInformation)...
[17-05-14 08:57:54.600]Info:[Build Document.CompilePhaseHandler.RestApiDocumentProcessor]Building 4 file(s) in RestApiDocumentProcessor(BuildRestApiDocument=>ValidateRestApiDocumentMetadata=>ApplyOverwriteDocumentForRestApi)...
[17-05-14 08:57:54.602]Info:[Build Document.CompilePhaseHandler.TocDocumentProcessor]Building 3 file(s) in TocDocumentProcessor(BuildTocDocument)...
[17-05-14 08:57:54.629]Warning:[Build Document.CompilePhaseHandler.TocDocumentProcessor.Prebuild.BuildTocDocument](toc.yml)Unable to find either toc.yml or toc.md inside obj/api/. Make sure the file is included in config file docfx.json!
[17-05-14 08:57:55.639]Info:[Build Document.CompilePhaseHandler.ManagedReferenceDocumentProcessor.Build.BuildManagedReferenceDocument](specs/petstore-operations.md#L1)Invalid yaml header.
[17-05-14 08:57:55.694]Info:[Build Document.CompilePhaseHandler.RestApiDocumentProcessor.Build.BuildRestApiDocument](specs/petstore-operations.md#L1)Invalid yaml header.
[17-05-14 08:57:56.207]Info:[Build Document.LinkPhaseHandler.Apply Templates]Applying templates to 11 model(s)...
[17-05-14 08:57:56.519]Info:[Build Document]XRef map exported.
[17-05-14 08:57:56.619]Info:[Postprocess.HandlePostProcessors.Processing ExtractSearchIndex]Extracting index data from 7 html files
[17-05-14 08:57:56.666]Info:[Postprocess]Manifest file saved to manifest.json.
[17-05-14 08:57:56.666]Info:Completed building documents in 2824.5081 milliseconds.
[17-05-14 08:57:56.666]Info:[Apply Theme]Theme is applied.
[17-05-14 08:57:56.666]Info:Completed executing in 5018.2267 milliseconds.


Build succeeded with warning.
[17-05-14 08:57:56.666]Warning:Cache for H:/docfx-test/src/SampleClass1/SampleClass1.sln in H:\docfx-test\src\SampleClass1\obj\xdoc\cache\final\1563805426 is corrupted
[17-05-14 08:57:56.666]Warning:No metadata is generated for CatLibrary,BaseClass,SomeVbClass.
[17-05-14 08:57:56.666]Warning:[Build Document.CompilePhaseHandler.TocDocumentProcessor.Prebuild.BuildTocDocument](toc.yml)Unable to find either toc.yml or toc.md inside obj/api/. Make sure the file is included in config file docfx.json!
        3 Warning(s)
        0 Error(s)

H:\docfx-test>pause
Press any key to continue . . .
```

**Expected Result**
API Docs generation.

**Actual Result**
 *Cache is corrupted*

**Comment by roji** [comment]
Ran into this and solved it by following #1628 (i.e. install 2015 build tools).

**Comment by FranklinTimmy** [comment]
@roji Yep that seemed to have solved the problem. Thank you!
